app/api/routes.py

```
@@ -36,10 +36,38 @@ def get_engine(request: Request) -> TradingEngine:
+            validate_symbol(s)
+        # 대문자 정규화 (미국 심볼)
+        normalized = [s.upper() if s.isalpha() else s for s in cleaned]
+        # 중복 제거 (순서 유지)
+        seen: set[str] = set()
+        unique: list[str] = []
+        for s in normalized:
+            if s not in seen:
+                seen.add(s)
+                unique.append(s)
+        if len(unique) > _MAX_SYMBOLS:
+            raise ValueError(f"심볼은 최대 {_MAX_SYMBOLS}개까지 가능합니다 (입력: {len(unique)}개)")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)